teavm-classlib/src/main/java/org/teavm/classlib/java/util/TTreeMap.java

```diff
@@ -37,12 +37,12 @@ public TreeNode(K key) {
                 if (right.factor() < 0) {
                     right = right.rotateRight();
                 }
-                return rotateLeft();
+                return this;
```

Comment:
You can't call rotateLeft() or rotateRight() without assigning the result to the new root of the rotation otherwise it will result in the new root of the rotation to be removed.
This code by itself is not necessarily a problem.  It just place a condition, now, on the use of balance() that any call to balance() must assign the return value to the new root, and there were calls that did this (that have been removed). 

These changes do break balancing but it didn't look like there was a simple way to keep balance working while still making the tree function correctly without rewriting the whole thing.